src/core/utils/useWorkUrl.ts

```diff
@@ -0,0 +1,87 @@
+import { useMemo } from "react";
+import { useComplexSearchWithPaginationQuery } from "../dbc-gateway/generated/graphql";
+import { useUrls } from "./url";
+import { constructMaterialUrl } from "./helpers/url";
+import { Pid, WorkId } from "./types/ids";
+import { getMaterialTypes } from "./helpers/general";
+import { Manifestation } from "./types/entities";
+
+type UseWorkUrlParams = {
+  identifier?: string | null; // ISBN for digital materials
+  pid?: Pid | null; // PID for physical materials
+  materialType?: string | null;
+};
+
+type UseWorkUrlResult = {
+  workUrl: URL | null;
+  isLoading: boolean;
+};
+
+const useWorkUrl = ({
+  identifier,
+  pid,
+  materialType
+}: UseWorkUrlParams): UseWorkUrlResult => {
+  const u = useUrls();
+  const materialUrl = u("materialUrl");
+
+  // If we have a pid, we can construct the URL directly (physical path)
+  const physicalWorkUrl = useMemo(() => {
+    if (pid) {
+      const workId = `work-of:${pid}` as WorkId;
+      return constructMaterialUrl(
+        materialUrl,
+        workId,
+        materialType || undefined
+      );
+    }
+    return null;
+  }, [pid, materialUrl, materialType]);
+
+  // Only do the GraphQL query if we don't have a pid but have an identifier
+  const shouldQueryByIdentifier = !pid && Boolean(identifier);
+
+  const { data, isLoading } = useComplexSearchWithPaginationQuery(
+    {
+      cql: `term.isbn=${identifier}`,
+      offset: 0,
+      limit: 1,
+      filters: {}
```

Comment:
Might be a stupid question, but what about the global filters? Such as blacklisted branches?

Reply:
Good question! In this case we're not filtering by branch because:

- The user already has this loan/reservation - we're just looking up the work ID to construct a URL to the work page
- We're not showing search results or availability, just linking to an existing item
- The work exists globally regardless of branch holdings

It is a valid concern though, if the hook were to be used in places where the URL would have to be specific to a branch.